### PR TITLE
Hats bug fix and conformation to other transforms

### PR DESCRIPTION
**Pull request by fabhertz95**

### CODE REVIEW

File: tonic/functional/to_averaged_timesurface.py [hunk 1]
```diff
@@ -94,8 +69,6 @@ def to_averaged_timesurface(
         temporal_window (int): how far back to look for past events for the time averaging. Expressed in microseconds.
```
Comment (biphasic) [inline tonic/functional/to_averaged_timesurface.py:69]: I think this should be time_window

File: tonic/functional/to_averaged_timesurface.py [hunk 2]
```diff
@@ -105,65 +78,30 @@ def to_averaged_timesurface(
...
-    )
-
-    if not use_joblib:
-        histograms = np.stack([get_cell_histogram(c) for c in range(n_cells)])
-    else:
-        histograms = np.stack(
-            Parallel(n_jobs=num_workers)(
-                delayed(get_cell_histogram)(c) for c in range(n_cells)
-            )
-        )
```
Comment (biphasic) [inline tonic/functional/to_averaged_timesurface.py:168]: So this is now not using joblib anymore?

File: tonic/transforms.py [hunk 3]
```diff
@@ -686,30 +686,28 @@ class ToAveragedTimesurface:
         sensor_size: a 3-tuple of x,y,p for sensor_size
         cell_size (int): size of each square in the grid
         surface_size (int): has to be odd
-        temporal_window (float): how far back to look for past events for the time averaging
+        time_window (float): how far back to look for past events for the time averaging
         tau (float): time constant to decay events around occuring event with.
         decay (str): can be either 'lin' or 'exp', corresponding to linear or exponential decay.
         num_workers (int): number of workers to be deployed on the histograms computation. When >1, joblib is required.
```
Comment (biphasic) [inline tonic/transforms.py:692]: think this line needs to go too

**Review by biphasic — CHANGES_REQUESTED** [review]
I'm curious to hear why you removed the parallel processing! I did see that the tests with multithreading lagged quite a bit that's why I removed the multiprocessing parameter from the tests in the past

**Comment by fabhertz95** [comment]
> I'm curious to hear why you removed the parallel processing! I did see that the tests with multithreading lagged quite a bit that's why I removed the multiprocessing parameter from the tests in the past

Basically, with the latest fine-tuning of the code in which everythin is vectorized through NumPy, the joblib addition was just a waste of code :) Now it's much cleaner, in my opinion. I will apply the patches that you requested!

**Pull request merged**